### DISCUSSION TITLE
How can I add more than 10 tags?

### DISCUSSION THREAD

**Discussion by pjotrek-b** [minio/minio#19463] [Q&A] [answered]
I'm evaluating Minio for long-term, large-scale storage setups, and would like to excessively make use of the metadata/tagging fields.

How can I increase the default limit of max. 10 tags - and what would the consequences be?

Thanks in advance for any information! :smile:

**Comment by harshavardhana** [comment, marked as answer]
You cannot - its an S3 API limit

**Reply by pjotrek-b** [reply]
Hm. Interesting.
I'm absolutely new to Object Storages and S3 capabilities.

However, reading the paragraph "[Why should I use object tags? (Amazon AWS Blog)](https://aws.amazon.com/blogs/storage/simplify-your-data-lifecycle-by-using-object-tags-with-amazon-s3-lifecycle/), I would get the impression that 10 tags with a very limited character set allowed would be insufficient for what they describe.

Quote:

> "The tags can be used to manage and control access, set up lifecycle rules, customize S3 Storage Class Analysis, and filter CloudWatch metrics. You can think of the bucket as a data lake, and use tags to create a taxonomy of the objects within the lake. This is more flexible than using the bucket and prefixes, and allows you to make semantic-style changes without renaming, moving, or copying objects."

That sounds exactly like a tagging-use I'm looking for: Creating taxonomies and annotating Objects.
So I am surprised to see max. 10 tags with a very limited set of characters allowed.
Am I missing something here?


Thanks, and grateful for any insights. :smile:

**Reply by harshavardhana** [reply]
https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html

**Reply by pjotrek-b** [reply]
I've desperately, and seriously tried my best to find answers to "are these real-technical- or arbitrarily-chosen-limits"?

Yes, Amazon has defined that for their services, yet if it were merely a quick edit of an integer in a config/header file - I'd be very professionally interested to give this a try.

Has noone ever wanted to have more than 10 tags?
Am I using tags "wrong"? :face_with_diagonal_mouth: 

Grateful for any insights! :chicken: :egg: 
Thank you.

**Reply by meyraa** [reply]
Does this mean, if I am using a self hosted minio with mc I have no tag limits?

**Reply by pjotrek-b** [reply]
@meyraa: AFAIK unfortunately, not.

I have set up a local, self-hosted minio and tried the Python API as well as mc to set tags (and metadata fields) - and the limitations (amount, length and charset) are enforced by minio's code itself.

Depending on how these limitation-checks are implemented though, it should be possible to patch minio's code to allow unlimited (or at least more?) tags-and-metadata fields (I hope). I haven't had the time to find the corresponding parts of minio's code yet.

Interested to help? :grin: 

Of course this would "break" S3 compatibility - but since I don't care about S3, I don't care :wink:
